server/src/entities/classes/destroyables/movables/characters/Character.js

```diff
@@ -38,6 +39,11 @@ class Character extends Movable {
         // If any of the damage types are not blocked, the full damage is dealt.
         if (damage.canAffectTarget(this) === false) return;
 
+        if (this.curse !== undefined && this.curse !== null) {
+            if (this.curse instanceof Pacify) {
+                this.curse.remove();
+            }
+        }
```

Comment:
Upon looking how the magic effects work, I think this can be simplified as MagicEffect can be triggered by damage, but atm it only activates for enchantments (i.e. ward). Can just trigger it for curses to, then add the removal logic to the Pacify curse instead (also like ward).

Character.js
```js
onDamage(damage, source) {
    // Check if they have an enchantment or curse that responds to being damaged.
    if (this.enchantment !== null) {
        if (this.enchantment.onCharacterDamaged() === false) {
            return;
        }
    }

    if (this.curse !== null) {
        if (this.curse.onCharacterDamaged() === false) {
            return;
        }
    }

    // If the character is immune to all of the types of the incoming damage, ignore the damage completely.
    // If any of the damage types are not blocked, the full damage is dealt.
    if (damage.canAffectTarget(this) === false) return;

    ...
```

And add a override for the `onCharacterDamaged` method to Pacify:

MagicEffects.js
```js
class Pacify extends Curse {
    onCharacterAttack() {
        return false;
    }

    onCharacterDamaged() {
        this.remove();
        return true;
    }
}
```